projects/core/components/data-list/option/option.component.ts

```diff
@@ -86,7 +86,7 @@ export class TuiOptionComponent<T = unknown> implements OnDestroy {
 
     @HostListener('click')
     onClick(): void {
-        if (this.host && this.value !== undefined) {
+        if (this.host && this.value !== undefined && !this.disabled) {
```

Comment:
I don't think this is necessary, it is covered by pointer events and native disable of the button.

Reply:
fixed